Source/WebKit/UIProcess/AuxiliaryProcessProxy.cpp

```diff
@@ -45,10 +45,54 @@
 #import <pal/spi/ios/MobileGestaltSPI.h>
 #endif
 
+#if PLATFORM(VISION) && defined(__has_include) && __has_include(<libkern/OSThermalNotification.h>)
+#import <libkern/OSThermalNotification.h>
+#import <notify.h>
+#endif
+
 namespace WebKit {
 
+static Seconds adjustedTimeoutForThermalState(Seconds timeout)
+{
+#if PLATFORM(VISION) && defined(__has_include) && __has_include(<libkern/OSThermalNotification.h>)
+    static int thermalState = kOSThermalPressureLevelNominal;
+    static dispatch_once_t once = 0L;
+
+    dispatch_once(&once, ^{
+        int localToken;
+        uint32_t result = notify_register_dispatch(kOSThermalNotificationPressureLevelName, &localToken, dispatch_get_main_queue(), ^(int token) {
```

Comment:
Any reason you chose this level of thermal notification instead of NSProcessInfo like in https://commits.webkit.org/267480@main?